core/graph/src/main/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandler.java

```diff
@@ -46,97 +47,146 @@ protected OtherGraphExporter createExporter(final ExportToOtherGraph export, fin
-                }
-            }
-            if (exportSchema != null) {
-                if (schema == null) {
-                    schema = exportSchema;
-                } else {
-                    // delete the old schema id as we are about to modify the schema
-                    schema = new Schema.Builder()
-                            .merge(schema)
-                            .id(null)
-                            .merge(exportSchema)
-                            .build();
+    private Schema resolveSchema(final Schema exportSchema, final List<String> exportParentSchemaIds, final GraphLibrary graphLibrary, final Schema storeSchema) {
+        Schema schema = null;
+        if (exportParentSchemaIds != null) {
+            if (exportParentSchemaIds.size() == 1) {
+                schema = graphLibrary.getSchema(exportParentSchemaIds.get(0));
+            } else {
+                final Schema.Builder schemaBuilder = new Schema.Builder();
+                for (final String id : exportParentSchemaIds) {
+                    schemaBuilder.merge(graphLibrary.getSchema(id));
                 }
+                schema = schemaBuilder.build();
             }
-            if (null == schema) {
-                schema = storeSchema;
+        }
+
+        if (exportSchema != null) {
+            if (schema == null) {
+                schema = exportSchema;
+            } else {
+                // delete the old schema id as we are about to modify the schema
+                schema = new Schema.Builder()
+                        .merge(schema)
+                        .id(null)
+                        .merge(exportSchema)
+                        .build();
             }
         }
+        if (null == schema) {
+            schema = storeSchema;
+        }
+        return schema;
+    }
 
+    private Graph getGraphWithLibraryAndID(final String exportGraphId, final GraphLibrary graphLibrary) {
+        // If the graphId exists in the graphLibrary then just use it
         return new Graph.Builder()
                 .graphId(exportGraphId)
                 .library(graphLibrary)
+                .build();
+    }
+
+    private Graph getGraphWithoutLibrary(final String exportGraphId, final StoreProperties exportStoreProperties, final StoreProperties storeStoreProperties, final Schema exportSchema, final Schema storeSchema) {
+        // No store graph library so we create a new Graph
+        final Schema schema = (null == exportSchema) ? storeSchema : exportSchema;
+        final StoreProperties properties = (null == exportStoreProperties) ? storeStoreProperties : exportStoreProperties;
+        return new Builder()
+                .graphId(exportGraphId)
                 .addSchema(schema)
-                .storeProperties(storeProperties)
+                .storeProperties(properties)
                 .build();
     }
 
+    private void validationHandling(final Store store, final ExportToOtherGraph<?> export) {
```

Comment:
Could we add all the validate to the validationHandling to just make one validate method?
This would also avoid the use of calling export.getX() twice as this is already done in the createGraph method once.